shell/common/dart_native_benchmarks.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/shell.h"
+
+#include "flutter/benchmarking/benchmarking.h"
+#include "flutter/shell/common/thread_host.h"
+#include "flutter/testing/dart_isolate_runner.h"
+#include "flutter/testing/fixtures_base.h"
+#include "flutter/testing/testing.h"
+#include "runtime/dart_vm_lifecycle.h"
+
+namespace flutter::testing {
+
+class DartNativeBenchmarks : public FixturesBase, public benchmark::Fixture {
+ public:
+  DartNativeBenchmarks() : FixturesBase() {}
+
+  void SetUp(const ::benchmark::State& state) {}
+
+  void TearDown(const ::benchmark::State& state) {}
+
+  void Wait() { latch_.Wait(); }
+
+  void Signal() { latch_.Signal(); }
+
+ private:
+  fml::AutoResetWaitableEvent latch_;
+
+  FML_DISALLOW_COPY_AND_ASSIGN(DartNativeBenchmarks);
+};
+
+BENCHMARK_F(DartNativeBenchmarks, DartNativeMessages)(benchmark::State& st) {
```

Comment:
For new benchmarks in this harness, it may be a good idea to add a docstring that explains what the benchmark is trying to measure as well as a more elaborate name. Right now, "DartNativeMessages" is pretty vague. I expected this case to measure the time it takes to make a call from Dart code to native code repeatedly. Instead, this benchmark measures time to first Dart message from a newly created isolate in a brand new VM.

Reply:
So, something like, "TimeToFirstNativeMessageFromIsolateInNewVM". I guess its verbose but gets the job done.

Reply:
Done. I also want to measure sending multiple messages from dart to native. I've added another test for that as well.